src/_pytest/tmpdir.py

```diff
@@ -31,10 +35,14 @@ class TempPathFactory:
     _given_basetemp = attr.ib(type=Optional[Path])
     _trace = attr.ib()
     _basetemp = attr.ib(type=Optional[Path])
+    _retention_count = attr.ib(type=int)
+    _retention_policy = attr.ib(type=str)
```

Comment:
```suggestion
    _retention_policy = attr.ib(type=Literal["none", "failed", "all"])
```

We should probably make a `TypeAlias` for that and reuse it everywhere it is declared.

Reply:
I made this.
```Python
RetentionType = Literal["all", "failed", "none"]
```